File: src/main/java/um/haberes/core/model/view/CodigoPeriodo.java
package um.haberes.core.model.view;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.annotations.Immutable;

@Entity
@Table(
        name = "vw_codigo_periodo",
        uniqueConstraints = {@UniqueConstraint(columnNames = {"anho", "mes", "codigoId"})}
)
@Immutable
@Getter
@Setter
@EqualsAndHashCode
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class CodigoPeriodo {

    @Id
    private Integer codigoId = null;

    private Integer anho = null;

    private Integer mes = null;
}

File: src/main/java/um/haberes/core/kotlin/model/view/CodigoPeriodo.kt
package um.haberes.core.kotlin.model.view

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint
import org.hibernate.annotations.Immutable

@Entity
@Table(name = "vw_codigo_periodo",
    uniqueConstraints = [UniqueConstraint(columnNames = ["anho", "mes", "codigoId"])])
@Immutable
data class CodigoPeriodo(

    @Id
    var codigoId: Int? = null,

    var anho: Int? = null,
    var mes: Int? = null

)
